examples/Example/src/prot_snap/index.tsx

```diff
@@ -0,0 +1,347 @@
+import React, { useState } from 'react';
+import { Alert, StyleSheet, useWindowDimensions } from 'react-native';
+import { GestureMonitor, Gesture } from 'react-native-gesture-handler';
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withTiming,
+  runOnJS,
+} from 'react-native-reanimated';
+import { useAnimatedGesture } from '../useAnimatedGesture';
+
+const filters = ['red', 'green', 'blue', 'yellow', 'orange', 'cyan'];
+const MAX_VIDEO_DURATION = 60000;
+
+export default function Home() {
+  const filter = useSharedValue(0);
+  const filterOffset = useSharedValue(0);
+  const zoom = useSharedValue(1);
+  const [scale, setScale] = useState(1);
+  const [selectedFilter, setSelectedFilter] = useState(0);
+  const [recording, setRecording] = useState(false);
+  const [remainingTime, setRemainingTime] = useState(MAX_VIDEO_DURATION);
+  const [recordingInterval, setRecordingInterval] = useState(-1);
+
+  const filtersPanGesture = Gesture.pan()
+    .setOnUpdate((e) => {
+      'worklet';
+      filter.value =
+        filter.value + (filterOffset.value - e.translationX) * 0.01;
+      filterOffset.value = e.translationX;
+
+      runOnJS(updateSelectedFilter)();
+    })
+    .setOnEnd(() => {
+      'worklet';
+      filterOffset.value = 0;
+      runOnJS(stopFilterScroll)();
+    });
+
+  const buttonTapGesture = Gesture.tap()
+    .setMaxDistance(3)
+    .setOnEnd((e, success) => {
+      'worklet';
+      if (success) runOnJS(takePhoto)();
+    });
+
+  const buttonDoubleTapGesture = Gesture.tap()
+    .setTapCount(2)
+    .setMaxDistance(3)
+    .setPriority(1)
+    .setOnEnd((e, success) => {
+      'worklet';
+      if (success) {
+        runOnJS(takeSeries)();
+      }
+    });
+
+  const buttonPanGesture = Gesture.pan()
+    .addSimultaneousGesture(filtersPanGesture)
+    .setOnUpdate((e) => {
+      'worklet';
+      if (recording) {
+        if (e.velocityY < 0) {
+          zoom.value = zoom.value * 1.05;
+        } else if (e.velocityY > 0) {
+          zoom.value = zoom.value * 0.95;
+        }
```

Comment:
Do you want to avoid `e.velocityY === 0`? If not, you can use ternary here as well.

Reply:
Yes, I wanted to avoid `e.velocityY === 0`, it may happen when you drag slowly and I figured I would do nothing in this case since it is just a quick demo.